### PR TITLE
Add to _check() API to accept a dict of config options for Checkers.

### PR DESCRIPTION
**Pull request by nigef**
We can now pass a keyword argument, a dict of config options, to `check_all()` or `check_errors()`.

An example of the dict of config options is:
```python
CONFIG = {
    # [ELIF]
    'max-nested-blocks': 3,
    # [FORMAT]
    'max-line-length': 80,
    # [FORBIDDEN IMPORT]
    'allowed-import-modules': ['doctest', 'unittest', 'hypothesis', 'python_ta'],
    # [FORBIDDEN IO]
    'allowed-io': None,
    # [MESSAGES CONTROL]
    'disable': ['R0401', 'R0901', 'R0903', 'R0904', 'R0911', 'R0916', 'W0402', 
                'W0403', 'W0410', 'W1501', 'W1502', 'W1505', 'E1300', 'E1301', 
                'E1302', 'E1304', 'W1300', 'W1301', 'W1302', 'W1304', 'E1124', 
                'E1125', 'E1129', 'E1132', 'W1402', 'W0105', 'E1303', 'W1306', 
                'W1307', 'E0116', 'E0114', 'E0112', 'E0115', 'E0106', 'E0113', 
                'E0111', 'E0105', 'E0100', 'E0117', 'W0150', 'W0120', 'W0124', 
                'W0108', 'W0123', 'W0122', 'W0110', 'C0122', 'C0200', 'W0141', 
                'W0640', 'W0623', 'W0614', 'W0604', 'W0603', 'W0602', 'W0601', 
                'E0604', 'E0603', 'E1200', 'E1201', 'E1202', 'W1201', 'E1205', 
                'E1206', 'similarities', 'newstyle', 'python3', 'W0512', 
                'C0403', 'C0401', 'C0402', 'E1701', 'E1700', 'W0332', 'C0327', 
                'C0328', 'E0202', 'E0241', 'E0704', 'W0211', 'W0232', 'W0511', 
                'R0204', 'C0303', 'W0231']
}
```

### CODE REVIEW

**Comment by david-yz-liu** [comment]
@nigef this is very nice (good work using the pylint API). However, ideally it would be the case that when the user passes in a dictionary, we still use the default .pylintrc file that ships with python_ta, only overriding the config fields with the key-value pairs from the dictionary.

**Comment by nigef** [comment]
I'm quite sure the defaults are set first. In [the `set_option` method](https://github.com/PyCQA/pylint/blob/master/pylint/config.py#L739) in `config.py`

File: python_ta/__init__.py [hunk 1]
```diff
@@ -67,13 +67,18 @@ def _load_pylint_plugins(current_reporter, local_config, pep8):
...
-            linter.global_set_option(key, local_config[key])
-    elif local_config != '':
+    if isinstance(local_config, str) and local_config != '':
+        # Use config file at the specified path instead of the default.
         linter.read_config_file(local_config)
     else:
+        # Use default config file in the python_ta package.
         linter.read_config_file(os.path.join(os.path.dirname(__file__), '.pylintrc'))
+
+    # Override part of the default config, with a dict of config options.
```
Comment (david-yz-liu) [inline python_ta/__init__.py]: Good stuff. I'd like to see this block go inside the previous `else`, though.

Comment (nigef) [inline python_ta/__init__.py]: done 👍

**Pull request merged**